### PR TITLE
Add Network Selector

### PR DESCRIPTION
**Pull request by radchukd**
Addresses #211

### CODE REVIEW

File: frontend/src/components/models.ts
Comment (mds1) [inline frontend/src/components/models.ts:28]: I like this layout for defining support chains. A few comments though:
- Can we make `rpcUrls`, `blockExplorerUrls`, and `iconUrls` from plural/arrays into singular/strings? Not sure I see the reasoning for making them arrays at the moment
- Can we add a `logoURI: string;` field to the `nativeCurrency` object? This way we have the token symbol of the native currency also. For ETH we can use `'/tokens/eth.svg'` as the URI

File: frontend/src/layouts/NetworkDropdown.vue
Comment (mds1) [inline frontend/src/layouts/NetworkDropdown.vue:4]: Perhaps make this button a little more subtle
```suggestion
    color="primary"
    no-caps
    outline
    rounded
```

File: frontend/src/components/models.ts
Comment (mds1) [inline frontend/src/components/models.ts:31]:
```suggestion
    chainName: 'Mainnet',
```

File: frontend/src/components/models.ts
Comment (mds1) [inline frontend/src/components/models.ts:43]:
```suggestion
    chainName: 'Rinkeby',
```

File: frontend/src/components/models.ts
Comment (mds1) [inline frontend/src/components/models.ts:25]: Would be great if the `blockExplorerUrl: string` field was directly used within the `getEtherscanUrl()` method in the `utils.ts` file! That way when adding a network we can just do it all in this file

File: frontend/src/utils/utils.ts
Comment (mds1) [inline frontend/src/utils/utils.ts:21]: Can this method just return the `chainName` from the `supportedChains` list? Similar to the Etherscan comment above—that way when adding a network we can just do it all in the one file, and these methods will pull from there

Also maybe rename to just getChainName since we'll pull directly from that list

File: frontend/src/layouts/NetworkDropdown.vue
Comment (mds1) [inline frontend/src/layouts/NetworkDropdown.vue:2]: This format of `<q-btn>` and `<q-list>` works, but you could also use our own `<base-select>` component which might be a little cleaner / less code here.

**Review by mds1 — CHANGES_REQUESTED** [review]
Loving this, thank you @radchukd!

------------------------

Main thing to update (as mentioned in the comments) is to have the `getEtherscanUrl` and `getChainShortName` methods pull form the `supportedChains` array, so everything can be defined in one spot. Currently adding a new network would require adding something in a few spots

It also seems the dropdown items aren't centered vertically as shown below. One easy way to fix this would be to use the `BaseSelect` component we have instead of implementing the dropdown yourself

image

File: frontend/src/layouts/NetworkDropdown.vue
Comment (mds1) [reply frontend/src/layouts/NetworkDropdown.vue:4]: Or if you switch to the `BaseSelect` component `:outline="true"` and `:rounded="true"`, but it seems we'd have to add a prop for `rounded` as it's not currently in the `BaseSelect` component

File: frontend/src/components/models.ts
Comment (radchukd) [reply frontend/src/components/models.ts:28]: I took this interface out of [EIP](https://eips.ethereum.org/EIPS/eip-3085#specification).

Comment (mds1) [reply frontend/src/components/models.ts:28]: Oh! Didn't realize that. Ok, fine with me to leave this as-is then 👌

Comment (mds1) [reply frontend/src/components/models.ts:28]: Maybe throw in a comment with a link to that EIP if you don't mind!

File: frontend/src/utils/utils.ts
Comment (radchukd) [reply frontend/src/utils/utils.ts:21]: On [Tornado](https://app.tornado.cash/) they have short name in the selected field and long names in dropdown.
If we keep the names short, there wouldn't be a need for this method at all.

Comment (mds1) [reply frontend/src/utils/utils.ts:21]: Good data point. I was looking at https://gnosis-safe.io/app/#/ which only uses short names. 

Personally I think the short names are sufficiently descriptive and I'd be ok removing this

Comment (mds1) [reply frontend/src/utils/utils.ts:21]: Maybe we wait to see what @apbendi thinks

**Comment by radchukd** [comment]
I have remade this with BaseSelect and width shouldn't be a problem now, so I have removed getChainShortName.

**Comment by mds1** [comment]
Thanks! Just pushed two small commits to cleanup some styling on the network selector, and make sure the chainId is properly converted from decimal to hex. LGTM, just waiting on @apbendi to review before merging

**Review by mds1 — APPROVED** [review]

**Review by apbendi — CHANGES_REQUESTED** [review]
Overall this looks awesome @radchukd! Great work, and thanks for tackling it so quickly. Here are a few things that came up in my QAing:

1. When testing in a mobile-size (and, weirdly, only in mobile size) I got the following error when switching between networks:

Screen Shot 2021-08-20 at 1 35 15 PM

Screen Shot 2021-08-20 at 1 35 26 PM

2. When the browser is in a slightly smaller size, the selector is truncated. In general, the selector seems a bit wide, even in browser sizes where it is not truncated.

Screen Shot 2021-08-20 at 1 33 12 PM

3. Right now when the user is on an unsupported network, the selector is not shown. Not sure how hard this would be, but I think a better flow would be to show the selector with "⚠️ Unsupported" shown, then allow the user to select the drop down and select a valid network from the list. Thoughts?

Screen Shot 2021-08-20 at 1 34 26 PM

**Comment by radchukd** [comment]
Everything should be fine now.
I have also added isLoading ref to the wallet store, so the network wouldn't be unsupported(undefined) at first, instead, there would be a spinner.

**Review by apbendi — APPROVED** [review]
Looking and working great @radchukd. Awesome work. Thanks!

**Pull request merged**
